ui/src/filters/index.ts

```diff
@@ -27,9 +37,15 @@ Vue.filter('formatDuration', (from: Date, to: Date) => {
   return humanizeDuration(moment.duration(diff).asMilliseconds());
 });
 
-Vue.filter('formatEpochDuration', (from: number, to: number) => {
-  const f = new Date(Math.round(from / 1000 / 1000));
-  const t = new Date(Math.round(to / 1000 / 1000));
+Vue.filter('formatEpochDuration', (from: EventTime, to: EventTime) => {
+  const f = fromEventTimeToDate(from)
+  const t = fromEventTimeToDate(to)
   const diff = moment(t).diff(f);
   return humanizeDuration(moment.duration(diff).asMilliseconds());
 });
+
+export function fromEventTimeToDate(ev :EventTime): Date {
+  let secondsToMilis = ev.seconds * 1000;
+  let nanosecondsToMilis = ev.nanos / 1_000_000;
+  return new Date(secondsToMilis + nanosecondsToMilis);
+}
```

Comment:
There's [packages/common/src/time.ts](https://github.com/temporalio/sdk-typescript/blob/HEAD/packages/common/src/time.ts) in the TS SDK if you want a quick, tested conversion helper.

Reply:
Yeah it's overkill